src/System.Private.CoreLib/src/System/Threading/WaitSubsystem.HandleManager.Unix.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal static partial class WaitSubsystem
+    {
+        private static class HandleManager
+        {
+            public static IntPtr NewHandle(WaitableObject waitableObject)
+            {
+                Debug.Assert(waitableObject != null);
+
+                IntPtr handle = GCHandle.ToIntPtr(GCHandle.Alloc(waitableObject));
+                Debug.Assert(handle != new IntPtr(-1)); // SafeWaitHandle treats -1 and 0 as invalid
+                if (handle == IntPtr.Zero)
```

Comment:
Would it look better to just call RhHandleAlloc directly to not have these conversions between GCHandle and IntPtr?